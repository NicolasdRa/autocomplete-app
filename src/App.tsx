import { Search } from './components/Search/Search';
import { CustomTable } from './components/CustomTable/CustomTable';

function App() {
	return (
		<div className='App'>
			<h1>Autocomplete App</h1>
			<Search />
			<CustomTable />
		</div>
	);
}

export default App;
