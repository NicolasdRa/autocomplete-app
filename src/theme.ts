import { createTheme } from '@material-ui/core/styles';

export default createTheme({
	palette: {
		primary: {
			main: 'hsl(229, 25%, 31%)',
		},
		secondary: {
			main: 'hsl(349, 71%, 52%)',
		},
		common: {
			white: '#ffffff',
			grey: '#dde0e1',
			shadow: '#a3b0c7',
		},
		background: {
			default: 'white',
		},
	},

	typography: {
		fontFamily: ['Barlow Semi Condensed', 'sans-serif'].join(','),
	},

	overrides: {},

	breakpoints: {
		values: {
			xs: 0,
			sm: 375,
			md: 960,
			lg: 1366,
			xl: 1920,
		},
	},
});
