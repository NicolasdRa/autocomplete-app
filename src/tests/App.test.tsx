import ThemeProvider from '@material-ui/styles/ThemeProvider';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import App from '../App';
import theme from '../theme';

describe('tests on <App/>', () => {
	const wrapper = mount(
		<ThemeProvider theme={theme}>
			<App />
		</ThemeProvider>,
	);

	test('should render <App/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
