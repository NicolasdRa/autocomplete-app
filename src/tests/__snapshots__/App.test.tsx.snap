// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <App/> should render <App/> correctly 1`] = `
Array [
  <div
    className="makeStyles-root-3"
  >
    <header
      className="MuiPaper-root-18 MuiAppBar-root-7 MuiAppBar-positionStatic-11 MuiAppBar-colorPrimary-14 MuiPaper-elevation4-25"
    >
      <div
        className="MuiToolbar-root-46 MuiToolbar-regular-48 MuiToolbar-gutters-47"
      >
        <div
          className="MuiBox-root-50 MuiBox-root-51"
        >
          <a
            href="!#"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="logo"
              className="makeStyles-logo-6"
              src="logo-rounded.png"
            />
          </a>
        </div>
        <h6
          className="MuiTypography-root-52 makeStyles-title-5 MuiTypography-h6-62"
        >
          Autocomplete App
        </h6>
        <button
          aria-label="account of current user"
          className="MuiButtonBase-root-90 MuiIconButton-root-82 MuiIconButton-colorInherit-85"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label-89"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root-91"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-231"
          />
        </button>
      </div>
    </header>
  </div>,
  <div
    className="MuiBox-root-50 MuiBox-root-100 makeStyles-generalContainer-1"
  >
    <div
      className="MuiBox-root-50 MuiBox-root-241 makeStyles-skeleton-2"
    >
      <span
        className="MuiSkeleton-root-242 MuiSkeleton-text-243 MuiSkeleton-pulse-246"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root-242 MuiSkeleton-text-243"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root-242 MuiSkeleton-text-243 MuiSkeleton-wave-248"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
  </div>,
]
`;
